### PR TITLE
Avoid storing logs in Keeper containing unknown operation

### PR DESCRIPTION
**Pull request by antonio2368**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Keeper fix: avoid storing logs containing invalid operations so the instance doesn't end up in an invalid state.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Coordination/Changelog.h [hunk 1]
```diff
@@ -39,7 +39,7 @@ struct ChangelogRecordHeader
     ChangelogVersion version = CURRENT_CHANGELOG_VERSION;
     uint64_t index = 0; /// entry log number
     uint64_t term = 0;
-    nuraft::log_val_type value_type{};
+    int32_t value_type{};
```
Comment (antonio2368) [inline src/Coordination/Changelog.h:42]: https://github.com/eBay/NuRaft/pull/404

**Comment by robot-clickhouse-ci-1** [comment]
This is an automated comment for commit ab8365630b3ee50120e67664ca4ecbab1afcc4c3 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50751/ab8365630b3ee50120e67664ca4ecbab1afcc4c3/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by alesapin — APPROVED** [review]
Changes are good, but I think we need to add rolling update integration test.... Otherwise we will see something similar soon...

**Comment by antonio2368** [comment]
> Changes are good, but I think we need to add rolling update integration test.... Otherwise we will see something similar soon...

Yes, agree
Until we have those we pause with adding new write requests to Keeper

**Pull request merged**